### PR TITLE
For #11387 - Update changelog with an entry on the PromptFeature constructor removal

### PR DESCRIPTION
**Pull request by sarah541**
Update changelog.md in respect with removal of constructor from PromptFeature.kt

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by sarah541** [comment]
Associated change in PromptFeature.kt : https://github.com/mozilla-mobile/android-components/pull/11390

File: docs/changelog.md [hunk 1]
```diff
@@ -11,6 +11,10 @@ permalink: /changelog/
 * [Gecko](https://github.com/mozilla-mobile/android-components/blob/main/buildSrc/src/main/java/Gecko.kt)
 * [Configuration](https://github.com/mozilla-mobile/android-components/blob/main/.config.yml)
 
+* **feature-prompts**:
+  *Removes deprecated `constructor`
```
Comment (gabrielluong) [inline docs/changelog.md:15]:
```suggestion
  * Removes deprecated constructor in `PromptFeature`. 
```

File: docs/changelog.md [hunk 2]
```diff
@@ -11,6 +11,10 @@ permalink: /changelog/
...
+
+
```
Comment (gabrielluong) [inline docs/changelog.md:17]: nit: extra new line

**Review by Amejia481 — APPROVED** [review]

**Review by gabrielluong — APPROVED** [review]

File: docs/changelog.md (hunk 2 above)
Comment (sarah541) [reply docs/changelog.md]: Fixed

**Pull request merged**